src/patch.js

```diff
@@ -19,6 +14,44 @@ export default class Patch {
     return new Iterator(this)
   }
 
+  rebalance () {
```

Comment:
Another thing I was wondering is could we rebalance only a portion of the tree after a splice sequence... Maybe from the leftmost new node to the rightmost new node. Would shrink N dramatically in many cases but might be tricky to implement.